i3lock.c

```diff
@@ -2232,6 +2241,19 @@ int main(int argc, char *argv[]) {
             case 905:
                 no_verify = true;
                 break;
+            case 906:
+                api_enabled = true;
+                if (optarg == NULL && argv[optind] != NULL &&
```

Comment:
What do these checks against argv do? Shouldn't optarg contain the parameter we need?

Reply:
I set `--enable-api` flag with an **optional argument** which can be accesed with optarg, but not in all cases:
- [x] `i3lock --enable-api="/tmp/custom-path"`
- [ ] `i3lock --enable-api "/tmp/custom-path"`

These checks, taken from [here](https://stackoverflow.com/a/40595790), allows the last case to be handled correctly.

Reply:
I'd suggest making separate option --api-path then. Leave the argument parsing to getopt so we don't need to reinvente the wheels.

Reply:
Oks